extras/windows/packageWindows.sh

```diff
@@ -1,158 +1,113 @@
-    exit -1
-else
-    echo "Something went wrong."
-    exit -1
-fi
+BUILD_DATE="$(date +%Y%m%d)"
 
-# Check if we are running in the right place
+# Output configured variables
 
-if [[ ! -f "naev.6" ]]; then
-    echo "Please run from Naev root directory."
-    exit -1
-fi
+echo "SOURCE ROOT:  $SOURCEROOT"
+echo "BUILD ROOT:   $BUILDPATH"
+echo "NIGHTLY:      $NIGHTLY"
+echo "BUILD OUTPUT: $BUILDOUTPUT"
 
 # Rudementary way of detecting which environment we are packaging.. 
-# It works, and it should remain working until msys changes their naming scheme
-
-if [[ $PATH == *"mingw32"* ]]; then
-    echo "Detected MinGW32 environment"
-    ARCH="32"
-elif [[ $PATH == *"mingw64"* ]]; then
-    echo "Detected MinGW64 environment"
-    ARCH="64"
-else
-    echo "Welp, I don't know what environment this is... Make sure you are running this in an MSYS2 MinGW environment"
-    exit -1
-fi
+# It works (tm), and it should remain working until msys changes their naming scheme
 
-VERSION="$(cat $(pwd)/dat/VERSION)"
-BETA=false
-# Get version, negative minors mean betas
-if [[ -n $(echo "$VERSION" | grep "-") ]]; then
-    BASEVER=$(echo "$VERSION" | sed 's/\.-.*//')
-    BETAVER=$(echo "$VERSION" | sed 's/.*-//')
-    VERSION="$BASEVER.0-beta.$BETAVER"
-    BETA=true
+# Check version exists and set VERSION variable.
+
+if [ -f "$SOURCEROOT/dat/VERSION" ]; then
+    VERSION="$(<"$SOURCEROOT/dat/VERSION")"
 else
-    echo "could not find VERSION file"
+    echo "The VERSION file is missing from $SOURCEROOT."
     exit -1
 fi
-
-# Download and Install mingw-ldd
-
-echo "Update pip"
-pip3 install --upgrade pip
-
-echo "Install mingw-ldd script"
-pip3 install mingw-ldd
+if [[ "$NIGHTLY" == "true" ]]; then
+    export VERSION="$VERSION.$BUILD_DATE"
+fi
+SUFFIX="$VERSION-win64"
 
 # Move compiled binary to staging folder.
 
 echo "creating staging area"
-mkdir -p extras/windows/installer/bin
+STAGING="$SOURCEROOT/extras/windows/installer/bin"
+mkdir -p "$STAGING"
 
 # Move data to staging folder
 echo "moving data to staging area"
-cp -r dat/ extras/windows/installer/bin
+cp -r "$SOURCEROOT/dat" "$STAGING"
 
 # Collect DLLs
- 
-if [[ $ARCH == "32" ]]; then
-for fn in `mingw-ldd naev.exe --dll-lookup-dirs /mingw32/bin | grep -i "mingw32" | cut -f1 -d"/" --complement`; do
-    fp="/"$fn
-    echo "copying $fp to staging area"
-    cp $fp extras/windows/installer/bin
-done
-elif [[ $ARCH == "64" ]]; then
-for fn in `mingw-ldd naev.exe --dll-lookup-dirs /mingw64/bin | grep -i "mingw64" | cut -f1 -d"/" --complement`; do
-    fp="/"$fn
-    echo "copying $fp to staging area"
-    cp $fp extras/windows/installer/bin
-done
-else
-    echo "Aw, man, I shot Marvin in the face..."
-    echo "Something went wrong while looking for DLLs to stage."
-    exit -1
-fi
+echo "Populating staging area with Meson subprojects' DLLs (if any)"
+find "${SOURCEROOT}/subprojects" -iname "*.dll" -exec cp -v "{}" "$STAGING" ";"
```

Comment:
I think some of the subprojects might build more dlls than we use, not sure if there's a smarter way to do this though.
Maybe `ninja -t query naev.exe` would be helpful?

Reply:
Hey, that code's my fault! :)
Yes, we can instead pass these roots to the extract_dlls script, and have it add them to its search.
I'd expect a small slowdown, nothing noticeable given how much work the script's already doing. A bit more complexity too.

The Ninja approach is quite cool. In general, if Meson winds up as The build system it will open up technically superior ways of doing the work.